src/vs/workbench/contrib/tasks/common/taskConfiguration.ts

```diff
@@ -1743,7 +1743,7 @@ namespace TaskParser {
 					result.custom.push(customTask);
 				}
 			} else {
-				let configuredTask = ConfiguringTask.from(external, context, index, source);
+				let configuredTask = ConfiguringTask.from(external, context, index, source, testTaskDefinition);
```

Comment:
would be good to avoid this, but not sure of a way to do that because of the way `TaskDefinitionRegistry` gets created

https://github.com/microsoft/vscode/blob/df681dcbfed9e82192a8359f7aff869541be915d/src/vs/workbench/contrib/tasks/common/taskDefinitionRegistry.ts#L182

https://github.com/microsoft/vscode/blob/99065b484f7fadddec2ad7c89028d92787e10224/src/vs/workbench/contrib/tasks/common/taskConfiguration.ts#L1402

Reply:
I can't use my usual tricks of injecting into the ctor or hiding in a protected method as it's a namespace not a class here.

Is there anyway we could swap out the registry with a test version?

Reply:
yes by passing it in. not sure if that's any better though?

Reply:
going with that